pkg/k8shandler/forwarding.go

```diff
@@ -0,0 +1,211 @@
+package k8shandler
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/pkg/generators/forwarding"
+	"github.com/openshift/cluster-logging-operator/pkg/logger"
+)
+
+const (
+	internalOutputName       = "clo-default-output-es"
+	collectorSecretName      = "fluentd"
+	logStoreService          = "elasticsearch.openshift-logging.svc:9200"
+	defaultAppPipelineName   = "clo-default-app-pipeline"
+	defaultInfraPipelineName = "clo-default-infra-pipeline"
+	secureForwardConfHash    = "8163d9a59a20ada8ab58c2535a3a4924"
+)
+
+var (
+	outputTypes = sets.NewString(string(logging.OutputTypeElasticsearch), string(logging.OutputTypeForward))
+	sourceTypes = sets.NewString(string(logging.LogSourceTypeApp), string(logging.LogSourceTypeInfra))
+)
+
+func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config string, err error) {
+
+	switch clusterRequest.cluster.Spec.Collection.Logs.Type {
+	case logging.LogCollectionTypeFluentd:
+		break
+	default:
+		return "", fmt.Errorf("%s collector does not support pipelines feature", clusterRequest.cluster.Spec.Collection.Logs.Type)
+	}
+
+	clusterRequest.ForwardingRequest = clusterRequest.normalizeLogForwarding(clusterRequest.cluster.Namespace, clusterRequest.cluster)
+	generator, err := forwarding.NewConfigGenerator(clusterRequest.cluster.Spec.Collection.Logs.Type)
```

Comment:
do you need to do something with `err` here?

Reply:
right, "err" should probably be changed to "_". The clusterRequest.cluster.Spec.Collection.Logs.Type value is checked in line 30.

Reply:
added error check

Reply:
Isn't the error check already in line 34? Probably line 39 won't be executed.